src/view.c

```diff
@@ -19,6 +20,24 @@
 #define LAB_FALLBACK_WIDTH  640
 #define LAB_FALLBACK_HEIGHT 480
 
+bool
+view_matches_query(struct view *view, struct view_query *query)
+{
+	bool match = true;
+
+	const char *identifier = view_get_string_prop(view, "app_id");
+	if (match && query->identifier && identifier) {
+		match &= match_glob(query->identifier, identifier);
+	}
+
+	const char *title = view_get_string_prop(view, "title");
+	if (match && query->title && title) {
+		match &= match_glob(query->title, title);
+	}
+
+	return match;
```

Comment:
- I don't understand the first `match` condition (and the `&=` in there), it is always true. I don't particularly dislike it due to the provided symmetry between the two conditions though. So consider this a nitpick.
- I am not sure if we could somehow end up getting `view_get_string_prop()` return `NULL`. In that case we could wrongly match the query.

`view_get_string_prop()` may return:
- `xdg_toplevel->title`
- `xdg_toplevel->app_id`
- `xwayland_surface->title`
- `xwayland_surface->class`

I think we should either verify (and document) that wlroots will never have any of these being `NULL` or we should check against `NULL` and return `""` in `view_get_string_prop()`.